BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -343,7 +343,7 @@ public final class ConvivaAnalytics: NSObject {
 
     private func onPlaybackStateChanged(playerState: PlayerState) {
         // do not report any playback state changes while player isStalled except buffering
-        if isStalled && playerState != .CONVIVA_BUFFERING {
+        if !isSessionActive || isStalled && playerState != .CONVIVA_BUFFERING {
```

Comment:
Do you know which event triggers the `onPlaybackStateChanged` in the case of casting? 🤔 It sounds odd to me that we even get into here? 🤔

_Even this fixes the crash I have concerns that this fixes the root cause_ 🤔

Reply:
hello @stonko1994 thanks for your feedback. After this I went and took a deeper look on what is going on here.
What I noticed is that the `PlaylistTransitionEvent` is being triggered by the player after the user hits the Chromecast button. Inside the Conviva code, this event is at the same time triggering `PlaybackFinishedEvent` which derives on the rest of the problem here (trying to change the status when there is no session active).
However, all of this only happens when the player loads the source from the sourceConfig like this: `player.load(sourceConfig:sourceConfig)` and **never** happens when Instead it loads it from a `Source` object , created from the SourceFactory (`
 let source = SourceFactory.create(from: sourceConfig)
       player.load(source: source)`)

As I had a conversation with @luckygoyal-bitmovin , I understand this second option from `SourceFactory` is the recommended solution when using casting, so it looks like this change is not needed at all.
However, want to raise that this happens with the other approach, in case this could be a concern on certain scenarios.

Reply:
Thanks, @thomhs22 for the detailed write-up.

I think this is related to a known issue (_Internal issue number: PI-1405_)